lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -151,14 +151,55 @@ function makeComparison(name, actualResult, expectedResult) {
   };
 }
 
+/**
+ * Delete expectations that don't match environment criteria.
+ * @param {LocalConsole} localConsole
+ * @param {LH.Result} lhr
+ * @param {Smokehouse.ExpectedRunnerResult} expected
+ */
+function pruneExpectations(localConsole, lhr, expected) {
+  const userAgent = lhr.userAgent;
+  const userAgentMatch = /Chrome\/(\d+)/.exec(userAgent); // Chrome/85.0.4174.0
+  if (!userAgentMatch) throw new Error('Could not get chrome version.');
+  const actualChromeVersion = Number(userAgentMatch[1]);
+
+  /**
+   * @param {*} obj
+   */
+  function failsChromeVersionCheck(obj) {
+    if (!obj._minChromeMajorVersion) return false;
+    return actualChromeVersion < obj._minChromeMajorVersion;
+  }
+
+  /**
+   * @param {*} obj
+   */
+  function pruneNewerChromeExpectations(obj) {
+    for (const key of Object.keys(obj)) {
+      const value = obj[key];
+      if (!value || typeof value !== 'object') continue;
+      else if (failsChromeVersionCheck(value)) {
+        localConsole.log(`[${key}] failed chrome version check, pruning expectation: ${
+          JSON.stringify(value, null, 2)}`);
+        delete obj[key];
```

Comment:
I don't believe we clone the test expectations anywhere, so this will alter them for any other uses of them as well (e.g. in smokerider if it ever runs against two different browser versions). Probably not what is wanted.

Reply:
Cloning here is tricky because of regexs. I can't see a situation where we'd run smokehouse twice in the same process but for different browsers. At the very least, one would configure the browser to use with CHROME_PATH between multiple invocations.

Reply:
would this protect against this case?

https://github.com/GoogleChrome/lighthouse/blob/6c4bfeec80fcdd4f72541313bc3d994d707cbed0/lighthouse-cli/test/smokehouse/frontends/lib.js#L26

(if so, we could also add cloneDeep to smokehouse-bin.js)

Reply:
If we're worried about mutations the cloning should happen in `pruneExpectations`. It should use the lodash clone and return a new object.

Reply:
> there are only two valid entrypoints for smoke tests: `lib.js` and `bin.js`. The former is already cloning. The latter is a CLI entry point and _should never run more than once_, so state mutation isn't problematic (for the reasons we're discussing here).

and `node.js`! There's an ascii diagram and everything:

https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-cli/test/smokehouse/readme.md#pipeline

:)

I don't think it's worth sidestepping cloning, especially because we already have the dependency and it's fairly fast to clone since our expectation objects are so small. The failure mode OTOH is completely silent since smokehouse only asserts what's in the expectations, so it'll be difficult to notice that anything was removed from them and not put back.